src/main/java/randoop/generation/DemandDrivenInputCreator.java

```diff
@@ -0,0 +1,406 @@
+import randoop.util.SimpleArrayList;
+import randoop.util.SimpleList;
+
+/**
+ * A demand-driven approach to construct inputs. Randoop works by selecting a method and then trying
+ * to find inputs for that method. Ordinarily, Randoop works bottom-up: if Randoop cannot find
+ * inputs for the selected method, it gives up and selects a different method. This demand-driven
+ * approach works top-down: if Randoop cannot find inputs for the selected method, then it looks for
+ * methods that create values of the necessary type, and iteratively tries to call them.
+ *
+ * <p>The demand-driven approach implements the "Detective" component described by the ASE 2015
+ * paper <a href="https://people.kth.se/~artho/papers/lei-ase2015.pdf">"GRT: Program-Analysis-Guided
+ * Random Testing" by Ma et al.</a> .
+ */
+public class DemandDrivenInputCreator {
+  /**
+   * The principal set of sequences used to create other, larger sequences by the generator. New
+   * sequences are added on demand for creating object of missing types. Shared with {@link
+   * ComponentManager#gralComponents}.
+   */
+  private final SequenceCollection sequenceCollection;
+
+  /**
+   * A secondary sequence collection used to store sequences that are generated during the
+   * demand-driven input creation process. These sequences are added to the main sequence collection
+   * after the demand-driven input creation process is complete.
+   *
+   * <p>This is an optimization to reduce the search space for the missing types in the main
+   * sequence collection.
+   */
+  private final SequenceCollection secondarySequenceCollection;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that declare values of the
+   * exact type that was requested.
+   */
+  private boolean exactTypeMatch;
+
+  /**
+   * If true, {@link #createInputForType(Type)} returns only sequences that are appropriate to use
+   * as a method call receiver, i.e., Type.isNonreceiverType() returns false for the type of the
+   * variable created by the sequence.
+   */
+  private boolean onlyReceivers;
+
+  /** Constructs a new {@code DemandDrivenInputCreator} object. */
+  public DemandDrivenInputCreator(
+      SequenceCollection sequenceCollection, boolean exactTypeMatch, boolean onlyReceivers) {
+    this.sequenceCollection = sequenceCollection;
+    this.secondarySequenceCollection = new SequenceCollection();
+    this.exactTypeMatch = exactTypeMatch;
+    this.onlyReceivers = onlyReceivers;
+  }
+
+  /**
+   * Performs a demand-driven approach for constructing input objects of a target type, when the
+   * sequence collection contains no objects of that type.
+   *
+   * <p>This method processes all available constructors and methods to identify possible ways to
+   * create objects of the {@code targetType}. For each method or constructor, it attempts to
+   * generate a sequence by searching for necessary inputs from the provided sequence collection,
```

Comment:
What if the search is unsuccessful?  Does it give up, or does it recursively try, or what?